roachpb/data.go

```diff
@@ -275,23 +276,54 @@ func (t Timestamp) GoTime() time.Time {
 	return time.Unix(0, t.WallTime)
 }
 
+const (
+	checksumSize = 4
+	tagPos       = checksumSize
+	headerSize   = tagPos + 1
+)
+
+func (v *Value) checksum() uint32 {
+	if len(v.RawBytes) < checksumSize {
+		return 0
```

Comment:
what are values that return here? `nil` values (deletion tombstones)? Are any values in `(0, checksumSize)` possible in practice?